internal/abci/abci.go

```diff
@@ -1210,8 +1219,12 @@ func (a *AbciApp) validateProposalTransactions(ctx context.Context, txns [][]byt
 
 			// This block proposal may include transactions that did not pass
 			// through our mempool, so we have to verify all signatures.
-			if err = ident.VerifyTransaction(tx); err != nil {
-				return fmt.Errorf("transaction signature verification failed: %w", err)
+			if !a.TxSigVerified(txO.hash) {
+				if err = ident.VerifyTransaction(tx); err != nil {
```

Comment:
This is the headline change in the PR.  Validators processing block proposals would do repeated and expensive signature verification.